crates/autopilot/src/domain/settlement/surplus.rs

```diff
@@ -0,0 +1,120 @@
+//! Settlement surplus calculation
+
+use {
+    super::Encoded,
+    crate::domain::{auction, auction::order, eth},
+    num::BigRational,
+    number::conversions::big_rational_to_u256,
+    shared::conversions::U256Ext,
+    std::collections::HashMap,
+};
+
+/// Settlement surplus
+///
+/// Denominated in surplus tokens. Contains multiple values since settlement can
+/// have multiple orders with different tokens.
+pub struct Surplus {
+    surplus: HashMap<eth::TokenAddress, eth::TokenAmount>,
+}
+
+impl Surplus {
+    pub fn new(settlement: &Encoded) -> Self {
```

Comment:
Might be slightly nicer if this was a function on `Encoded` instead.
As is the function needs to use many methods and properties of the Settlement. If the function lived on `Encoded` instead there might be a chance to make some of these functions private instead or even delete them.

Reply:
done